dfget/core/core.go

```diff
@@ -209,6 +200,40 @@ func downloadFile(cfg *config.Config, supernodeAPI api.SupernodeAPI,
 	return err
 }
 
+func doDownload(cfg *config.Config, supernodeAPI api.SupernodeAPI,
+	register regist.SupernodeRegister, result *regist.RegisterResult, timeout time.Duration) error {
+	var getter downloader.Downloader
+	isBackDownload := false
+	if cfg.BackSourceReason > 0 {
+		getter = backDown.NewBackDownloader(cfg, result)
+		isBackDownload = true
+	} else {
+		printer.Printf("start download by dragonfly...")
+		getter = p2pDown.NewP2PDownloader(cfg, supernodeAPI, register, result)
+	}
+
+	err := downloader.DoDownloadTimeout(getter, timeout)
+	if err == nil {
+		// report finished task to supernode when success to download by dragonfly
+		reportFinishedTask(cfg, getter)
+		return nil
+	}
+
+	if isBackDownload {
+		return fmt.Errorf("failed to download file from source station: %v", err)
+	}
+
+	logrus.Errorf("failed to download by dragonfly: %v, and start try to download from source station", err)
+	printer.Printf("failed to download by dragonfly: %v, and start try to download from source station", err)
+
+	// try to download the file from the source directly
+	getter = backDown.NewBackDownloader(cfg, result)
+	if err := downloader.DoDownloadTimeout(getter, timeout); err != nil {
+		return err
```

Comment:
return fmt.Errorf("failed to download file from source: %v", err) .   Be compatible with the previous error.